### PR TITLE
New test files for various UTF-8 scenarios requiring the -eos flag

### PR DESCRIPTION
**Pull request by kenneth-olwing**
Your fixes for issue #83 (-eos flag) seems to work for my needs, so for completeness I fixed up the tests in the prior PR so at least they are there if you want them :-)

### CODE REVIEW

**Comment by shancock9** [comment]
Thanks for testing this with tidyall, and for the tests.  I'll get back to later you regarding the discussion of encoding/decoding issues.  Unfortunately the new test files are failing under windows ( see the above box ).  I am leaving on a short  trip right now, but I wonder if it is a line ending issue.  And I wonder if this would also cause a problem with tidyall under Windows, or does tidyall avoid this somehow.

**Comment by kenneth-olwing** [comment]
Hi,

Yes, I saw the mail from github on failing. And yes, it looks like a 
line ending problem.

TL;DR:

An informed guess is that core.autocrlf=true.

Can you influence that part of the build? At 
https://github.com/perltidy/perltidy/runs/4938409777?check_suite_focus=true, 
line 445, the checkout happens:

     "C:\Program Files\Git\bin\git.exe" checkout --progress --force 
refs/remotes/pull/85/merge

Switching this to:

     "C:\Program Files\Git\bin\git.exe" checkout -c core.autocrlf=false 
--progress --force refs/remotes/pull/85/merge

would probably fix this. Or, if the local/global config can be set for 
that, such as at line 22 when gc is turned off.

===

I've actually used Windows and Perl 5.30 when doing this, and testing 
works fine here.

The input has CRLF, while the output has plain LF. This is a basic 
problem to begin with. As with encoding, what I pass in should come 
back, e.g. line endings shouldn't magically change, ever (and in fact, 
this plagued tidyall too, until I submitted a PR on C::T for that :-) 
see https://github.com/houseabsolute/perl-code-tidyall/pull/97).

Now, in my case, we work on the same code from both Windows and Linux, 
with tidyall (perltidy and perlcritic) running on both and producing 
identical output. This is achieved by passing -ole=unix to perltidy in 
order to ensure we always have one standard - so, for me the flapping of 
line endings is not a problem as I force it to one known. Again though, 
this is old tried and known behavior so...

Notice I said that the *input* has CRLF. This input comes from files in 
the t directory which the tests read in :raw, i.e. no attempt to manage 
line endings. Also, in the repo they *are* stored with plain LF. But 
still, the CRLF shows up. It's likely that the git client on the test 
machine(s) are set up with 'core.autocrlf=true', and so git 'helpfully' 
converts line endings in files. That was a bad idea in WinZip and it's a 
bad idea in git - that it allows this is mindboggling. Again, what I put 
into a CM system should come back out exactly as they we're originally 
stored.

ken1

**Comment by shancock9** [comment]
Hi Ken,
You guessed right.  After a couple of bungled attempts it looks like the new updates are working under Windows on github/actions.  The trick was to include a file t/.gitattributes with a line '* -text' to prevent converting the LF's to CRLF's in text files.
Steve

**Comment by kenneth-olwing** [comment]
Thumbs up, glad it works. I never thought about .gitattributes.

It's sad though that a workaround is necessary for behavior that 
shouldn't have been there in the first place :-/. Oh well, reality isn't 
always what we want it to be.

Have a good one,

ken1
